backend/api/services/CreditTradeService.py

```diff
@@ -369,3 +373,73 @@ def get_allowed_statuses(credit_trade, request):
                 allowed_statuses.append("Approved")
 
         return allowed_statuses
+
+    @staticmethod
+    def dispatch_notifications(previous_status: CreditTradeStatus, credit_trade: CreditTrade):
+        notification_map = defaultdict(lambda: [])
+        government = Organization.objects.filter(type__type='Government').first()
+
+        StatusChange = namedtuple('StatusChange', ['new_status'])
+        ResultingNotification = namedtuple('ResultingNotification', ['recipient', 'notification_type'])
+
+        notification_map[StatusChange('Draft')] = [
+            ResultingNotification(credit_trade.initiator, NotificationType.CREDIT_TRANSFER_CREATED)
+        ]
+        notification_map[StatusChange('Submitted')] = [
+            ResultingNotification(credit_trade.initiator, NotificationType.CREDIT_TRANSFER_SIGNED_1OF2),
+            ResultingNotification(credit_trade.respondent, NotificationType.CREDIT_TRANSFER_SIGNED_1OF2)
+        ]
+        notification_map[StatusChange('Accepted')] = [
+            ResultingNotification(credit_trade.initiator, NotificationType.CREDIT_TRANSFER_SIGNED_2OF2),
+            ResultingNotification(credit_trade.respondent, NotificationType.CREDIT_TRANSFER_SIGNED_2OF2),
+            ResultingNotification(government, NotificationType.CREDIT_TRANSFER_SIGNED_2OF2),
+        ]
+        notification_map[StatusChange('Refused')] = [
+            ResultingNotification(credit_trade.initiator, NotificationType.CREDIT_TRANSFER_PROPOSAL_REFUSED),
+            ResultingNotification(credit_trade.respondent, NotificationType.CREDIT_TRANSFER_PROPOSAL_REFUSED)
+        ]
+        notification_map[StatusChange('Recommended')] = [
+            ResultingNotification(government, NotificationType.CREDIT_TRANSFER_RECOMMENDED_FOR_APPROVAL),
+        ]
+
+        notification_map[StatusChange('Not Recommended')] = [
+            ResultingNotification(government, NotificationType.CREDIT_TRANSFER_RECOMMENDED_FOR_DECLINATION),
+        ]
+
+
+        # figure this out. /approve method?
+
+        notification_map[StatusChange('Completed')] = [
+            ResultingNotification(credit_trade.initiator, NotificationType.CREDIT_TRANSFER_APPROVED),
+            ResultingNotification(credit_trade.respondent, NotificationType.CREDIT_TRANSFER_APPROVED),
+            ResultingNotification(government, NotificationType.CREDIT_TRANSFER_APPROVED),
+        ]
+
+        notification_map[StatusChange('Declined')] = [
+            ResultingNotification(credit_trade.initiator, NotificationType.CREDIT_TRANSFER_DECLINED),
+            ResultingNotification(credit_trade.respondent, NotificationType.CREDIT_TRANSFER_DECLINED),
+            ResultingNotification(government, NotificationType.CREDIT_TRANSFER_DECLINED),
+        ]
+
+        # handle rescinded
```

Comment:
Are these todo items? If they are please add TODO: on the comment so we can keep track of them.

Reply:
Yes they are. I will pull them out to my external todo list rather than commit them -- good catch.